File: programs/assignment3/ProgrammingAssignment3/ProgrammingAssignment3/Administrator.cpp
// COSC 1320 Summer 2015
// Name: Adrian
// Programming Assignment 3
// This is my own work; I will not post

#pragma once
#include <string>
#include "Administrator.h"
#include "HospitalEmployee.h"
using namespace std;

Administrator::~Administrator(void)
{
}
Administrator::Administrator() : HospitalEmployee('a' , "none" , 0){
	department = "none";
}
Administrator::Administrator(char newRole, string newName, int newID, string newDept) : HospitalEmployee(newRole , newName , newID){
	department = newDept;
}
void Administrator::setDept(string newDept){
	department = newDept;
}
string Administrator::getDept(){
	return department;
}
string Administrator::toString(){
	string str = " ";
	str += department;
	return (HospitalEmployee::toString() + str);
}
bool Administrator::equals(HospitalEmployee obj){
	if(obj.getID() == getID() && obj.getName() == getName() && obj.getRole() == getRole()){
		return true;
	} else {
		return false;
	}
}

File: programs/assignment1/ProgrammingAssignment1/src/programmingassignment1/Administrator.java
// COSC 1320 Summer 2015
// Name: Adrian
// Programming Assignment 1
// This is my own work; I will not post

package programmingassignment1;

/**
 * 
 * @author Adrian Davila
 */
public class Administrator extends Employee{
    String department;

    Administrator(){
        department = "";
    }
    Administrator(char newRole , String newName , int newID , String dpt){
        super(newRole , newName , newID);
        department = dpt;
        addEmployee((Employee)this);
    }

    public void deleteEmployee(){
        deleteEmployee(getName() , getRole());
    }

    // Mutators
    public void setDept(String newDept){
        department = newDept;
    }
    // Accessors
    public String getDept(){
        return department;
    }
    @Override
    public String toString(){
        return (super.toString() + " " + department);
    }
    @Override
    public String toString(boolean t){
        return (super.toString() + " Department: " + department);
    }
    @Override
    public boolean equals(Employee obj){
        if(obj == null){
            return false;
        }else if(getClass() != obj.getClass()){
            return false;
        }else{
            Administrator otherAdmin = (Administrator)obj;
            return (getName().equals(otherAdmin.getName()) && getRole() == otherAdmin.getRole());
        }
    }
}
